lib/src/main/java/com/auth0/msg/ClaimsValidator.java

```diff
@@ -0,0 +1,68 @@
+package com.auth0.msg;
+
+import com.auth0.jwt.exceptions.InvalidClaimException;
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import com.auth0.jwt.impl.PublicClaims;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public final class ClaimsValidator {
+    private final static Map<String, Claim> knownClaims = new HashMap<>();
+
+    static {
+        knownClaims.put("alg", new Claim("alg", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("cty", new Claim("cty", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("typ", new Claim("typ", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("kid", new Claim("kid", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("iss", new Claim("iss", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("sub", new Claim("sub", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.STRING));
+        knownClaims.put("exp", new Claim("exp", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.DATE));
+        knownClaims.put("nbf", new Claim("kid", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.DATE));
+        knownClaims.put("iat", new Claim("kid", Collections.<MessageType, List<Object>>emptyMap(), ClaimType.DATE));
+    }
+
+    public static boolean isKnownClaim (String name){
+        return knownClaims.containsKey(name);
+    }
+
+    public static boolean validate(String claimName, Object value, MessageType type){
+        Claim targetClaim = knownClaims.get(claimName);
+        switch (targetClaim.getType()) {
```

Comment:
As discussed, this logic doesn't work. We cannot check for instance of. We need to look at the value provided and try to convert that value somehow to the type and if we cannot then throw exception